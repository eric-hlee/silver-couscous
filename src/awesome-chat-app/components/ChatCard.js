import React from 'react'
import DeleteChat from '../containers/DeleteChat'

class ChatCard extends React.Component {

  render() {

    const style_chatCard = {
      backgroundColor: this.props.activeChat === this.props.id ? '#EAEEEE' : '#FFFFFF',
      padding: '15px',
      position: 'relative',
      overflow: 'auto',
    }

    const style_selectedChatCard = {
      width: 'calc(100% - 20px)',
      background: 'none',
      border: 'none', 
      padding: '0',
      font: 'inherit',
      color: 'black',
      fontSize: '16px',
      cursor: 'pointer',
      textAlign: 'center',
    }

    return (
      <div className="ChatCard" style={style_chatCard}>
        <button className="selectedChatCard" 
                id={this.props.id}
                onClick={this.props.onClick}
                style={style_selectedChatCard}>
          {this.props.lastMessage}
        </button>
        <DeleteChat id={this.props.id}/>
      </div>
    );
  }
}

export default ChatCard