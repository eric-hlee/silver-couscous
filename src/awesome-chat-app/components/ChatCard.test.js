import React from 'react'
import { shallow } from 'enzyme'
import ChatCard from './ChatCard'

describe('ChatCard', () => {
  it('should render self and subcomponents', () => {
    const wrapper = shallow(<ChatCard />)
    expect(wrapper).toMatchSnapshot()
  })
})
