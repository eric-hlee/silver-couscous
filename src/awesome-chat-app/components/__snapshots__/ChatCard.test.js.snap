// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatCard should render self and subcomponents 1`] = `
<div
  className="ChatCard"
  style={
    Object {
      "backgroundColor": "#EAEEEE",
      "overflow": "auto",
      "padding": "15px",
      "position": "relative",
    }
  }
>
  <button
    className="selectedChatCard"
    style={
      Object {
        "background": "none",
        "border": "none",
        "color": "black",
        "cursor": "pointer",
        "font": "inherit",
        "fontSize": "16px",
        "padding": "0",
        "textAlign": "center",
        "width": "calc(100% - 20px)",
      }
    }
  />
  <Connect(DeleteButton) />
</div>
`;
